template/filter.c.erb

```diff
@@ -137,106 +108,123 @@ filter_message_iterator_next_processing_reading(
     if (bt_message_get_type(upstream_message) != BT_MESSAGE_TYPE_EVENT) {
       btx_downstream_push_message(message_iterator_private_data,
                                   upstream_message);
-      continue;
-
-    }
-    /* Borrow the event message's event and its class */
-    const bt_event *event =
-        bt_message_event_borrow_event_const(upstream_message);
-    const bt_event_class *event_class = bt_event_borrow_class_const(event);
-
-    /* Call dispatcher or forward message downstream */
-    bool is_callback_called = false;
-    const char *class_name = bt_event_class_get_name(event_class);
-
-    {
-      name_to_dispatcher_t *s = NULL;
-      HASH_FIND_STR(common_data->name_to_dispatcher, class_name, s);
-      if (s) {
-        (*((dispatcher_t(*))(s->dispatcher)))(s->callbacks, common_data, upstream_message,
-                                              &is_callback_called);
-      }
-    }
-
-    {
-      const bt_stream_class *stream_class =
-          bt_event_class_borrow_stream_class_const(event_class);
-      const char *stream_class_name = bt_stream_class_get_name(stream_class);
-
-      name_to_dispatcher_t *s = NULL;
-      HASH_FIND_STR(common_data->name_to_matching_dispatcher, stream_class_name,
-                    s);
-      if (s) {
-        // is_callback_called will only be modified if at least one callback is
-        // called.
-        (*((dispatcher_t(*))(s->dispatcher)))(s->callbacks, common_data, upstream_message,
-                                              &is_callback_called);
-      }
-    }
-
-    if (is_callback_called) {
-      /* The message have been consumed, we can discard it */
-      bt_message_put_ref(upstream_message);
     } else {
-      btx_downstream_push_message(message_iterator_private_data,
-                                  upstream_message);
+      btx_call_dispatchers(upstream_message, message_iterator_private_data);
     }
   }
+}
 
-  /* Round Robin between Upstream MessageIterator */
-  message_iterator_private_data->head_mi =
-      message_iterator_private_data->head_mi->next;
-
-  if (message_iterator_private_data->queue) {
-    message_iterator_private_data->processing_state =
-        BTX_FILTER_PROCESSING_STATE_SENDING;
-    /* optimization */
-    return filter_message_iterator_next_processing_sending(
-        self_message_iterator, messages, capacity, count);
-  } else
-    return BT_MESSAGE_ITERATOR_CLASS_NEXT_METHOD_STATUS_AGAIN;
+static inline bool
+btx_closing_port(btx_message_iterator_t *message_iterator_private_data) {
+  bt_message_iterator_put_ref(
+      message_iterator_private_data->head_mi->message_iterator);
+  struct el_mi *tmp = message_iterator_private_data->head_mi;
+  CDL_DELETE(message_iterator_private_data->head_mi, tmp);
+  free(tmp);
+
+  return message_iterator_private_data->head_mi;
+}
+
+static inline bt_message_iterator_class_next_method_status
+filter_message_iterator_transition_from_reading_to_ending(
```

Comment:
Should rename to:
`filter_message_iterator_transition_from_reading_to_finalizing`